pipeline/transformer/transformer_mets/src/main/scala/uk/ac/wellcome/platform/transformer/mets/transformer/Mets.scala

```diff
@@ -0,0 +1,62 @@
+package uk.ac.wellcome.platform.transformer.mets.transformer
+
+import uk.ac.wellcome.models.work.internal._
+
+import scala.util.Try
+import cats.implicits._
+
+case class Mets(
+  recordIdentifier: String,
+  accessCondition: Option[String]
+) {
+
+  def toWork(version: Int): Try[UnidentifiedInvisibleWork] =
+    for {
+      maybeDigitalLocation <- digitalLocation
+      unidentifiableItem: MaybeDisplayable[Item] = Unidentifiable(
+        Item(locations = List(maybeDigitalLocation)))
+    } yield
+      UnidentifiedInvisibleWork(
+        version = version,
+        sourceIdentifier = sourceIdentifier,
+        workData(unidentifiableItem)
+      )
+
+  private def workData(unidentifiableItem: MaybeDisplayable[Item]) =
+    WorkData(
+      items = List(unidentifiableItem),
+      mergeCandidates = List(mergeCandidate))
+
+  private def mergeCandidate = MergeCandidate(
```

Comment:
Does it make a difference if things are `val`s or `def`s?

Reply:
In these sort of cases when no mutability / side effects:

`val` = Non lazy, stored in memory
`lazy val` = Lazy, stored in memory
`def` = Lazy, not stored in memory (recalculated each time needed)

Stylistically I quite like `def` here but don't think it matters much

Reply:
I suppose my concern would be that if it's a `def` it feels like we've said `There's a reason why this recalculates each time` is all.

Reply:
No, no reason. I tried replacing with val and it still works. The only thing is that, if they are vals, they need to be in a specific order so that each one is initialised when it's referenced by another one which is slightly annoying